app/controllers/course_user_data_controller.rb

```diff
@@ -136,13 +139,21 @@ def update
     end
   end
 
-  action_auth_level :destroy, :instructor
+  action_auth_no_course :destroy
   def destroy
-    @destroyCUD = @course.course_user_data.find(params[:id])
-    if @destroyCUD && @destroyCUD != @cud && params[:yes1] && params[:yes2] && params[:yes3]
-      @destroyCUD.destroy # awwww!!!
+    @course = Course.find_by(name: params[:course_name])
+    @cud = CourseUserDatum.unscoped.find(params[:id])
```

Comment:
Do we need to reload the course here?  It should already be defined.  Also this code doesn't make sure that @cud is actually in @course.